### ISSUE TITLE
Request alias is not identified in the container with RestfulStrategy

### ISSUE THREAD

**Issue by shadowhand** [thephpleague/route#37]
I am using `league/container` in my project and have this code to set up my container:

```php
$container->add('Router', function () use ($container) {
    $router = new RouteCollection($container);
    $router->setStrategy(new RestfulStrategy);
    // ... list of routes
    return $router;
});
```

And then this code to dispatch the request:

```php
$router     = $container->get('Router');
$request    = $container->get('Request');
$dispatcher = $router->getDispatcher();
$response   = $dispatcher->dispatch($request->getMethod(), $request->getPathInfo());
```

The problem is that when my controllers are called with `action($request)` the request is a new instance and not the request that was used to dispatch. I would suggest adding a method to `RestfulStrategy` that would allow the request name to be set:

```php
public function setRequest($request) { ... }
public function getRequest() { ... }

public function dispatch($controller, array $vars)
{
    // ...
    $request = $this->getRequest() ?: $this->getContainer()->get(...);
    // ...
}
```

This would allow me to do the following, without having to use a FQCN in my container:

```php
$container->set('DispatchStrategy', RestfulStrategy::class)
    ->withMethodCall('setRequest', ['Request']);
```

**Comment by shadowhand** [comment]
See also #26.

**Comment by philipobenito** [comment]
Yes the aliasing will be getting added to the strategy for v1 shortly, as for v2, it will be using PSR-7 for the request and response objects.

The reason this was emitted originally was so that the user could have the choice of how to build the request object. If it is done for you then it's limiting any manipulation of the request.

**Comment by philipobenito** [comment]
Closed as dupe of #26
